### ISSUE TITLE
autocomplete selection is broken on Firefox for Android

### ISSUE THREAD

**Issue by ofrias** [kraaden/autocomplete#104]
When running the autocomplete on Firefox for Android, setting the selection on the textbox does not work OK.

This happens because Firefox for Android fires the `oninput` event also when modifying the textbox value through Javascript (in the autocomplete demo with the code `inputfield.value = item.label` )

This was happening [in the past](https://github.com/kraaden/autocomplete/issues/99) too.

You can reproduce this with  [the demo](https://smartscheduling.com/en/documentation/autocomplete) and the current version of Firefox for Android (version 113.2.0).

**Comment by ofrias** [comment]
A possible fix is setting a flag before and after the `onSelect` call, for example:

```
                selecting = true;
                settings.onSelect(item, input);
                selecting = false;
```
And:
```
            if (! selecting) {
                fetch(0 /* Keyboard */);
            }
```

**Comment by denis-taran** [comment]
@ofrias Thank you for reporting this issue. It should be fixed now in the version 8.0.4
